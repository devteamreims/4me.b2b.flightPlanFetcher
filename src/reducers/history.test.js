import R from 'ramda';
import MockDate from 'mockdate';

import {
  markForHistory,
} from '../actions/history';

import {
  invalidIfplId,
} from '../actions/flightKeys';

import reducer, { maxHistoryLen } from './history';

describe('history', () => {
  describe('markForHistory', () => {
    test('should reject invalid arguments', () => {
      expect(() => markForHistory()).toThrowError(/argument/i);
      expect(() => markForHistory({})).toThrowError(/argument/i);
      expect(() => markForHistory({a: 'test'})).toThrowError(/argument/i);
      expect(() => markForHistory([{}, 2])).toThrowError(/argument/i);
    });

    test('should accept valid ifplIds list', () => {
      expect(() => markForHistory('ifplId1')).not.toThrow();
    });
  });
  describe('reducer', () => {
    test('should have an initial state', () => {
      expect(reducer(undefined, {type: '@@INIT'})).toEqual([]);
    });

    test('should mark specific keys are part of the history list', () => {
      const action = markForHistory('AA0');
      expect(reducer(undefined, action)).toMatchSnapshot();
    });

    test('should have a maximum size', () => {
      const getIfplId = i => `AA${i}`;

      let state = [];
      for(let i = 0; i < maxHistoryLen + 1; i++) {
        const ifplId = getIfplId(i);
        state = reducer(state, markForHistory(ifplId));
      }

      expect(state).not.toContain('AA0');
    });

    test('should remove keys when invalidated', () => {
      let state = ['AA0', 'AA1', 'AA2'];
      state = reducer(state, invalidIfplId('AA2'));
      expect(state).not.toContain('AA2');
    });
  });
});
