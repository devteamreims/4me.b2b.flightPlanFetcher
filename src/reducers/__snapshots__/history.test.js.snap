exports[`history reducer should mark specific keys are part of the history list 1`] = `
Array [
  "AA0",
]
`;
